src/Symfony/Component/HttpKernel/DataCollector/LoggerDataCollector.php

```diff
@@ -353,4 +359,28 @@ private function computeErrorsCount(array $containerDeprecationLogs): array
 
         return $count;
     }
+
+    private function extractDeprecationChannel(\Throwable $throwable = null): ?string
+    {
+        if (null === $throwable || !$this->projectDir) {
+            return null;
+        }
+
+        $appPrefix = $this->projectDir.'/src/';
+        $vendorPrefix = $this->projectDir.'/vendor/';
```

Comment:
We cannot assume that projects always look like this.

Reply:
I know. They just won't benefit of additional filtering which is fine imho. They'll just have the `php` channel like they do now.

Reply:
Maybe it would be better to match the path against `Composer\InstalledVersions::getInstalledPackages()` instead?

Reply:
Given the strict format used in the `trigger_deprecation()` contract, can't we extract the package name from the message instead? (Imho, we only need to be compatible with our own deprecation contract here)

Reply:
I guess we could, sure. But I'd like to have a general "Yes we want this" and "Yes the `channel` property is the correct way to go here" first 😄 I just hacked this together real quick to see if it was even working 😄

Reply:
I like the feature, but I'm indeed not sure if channel is the right property for this. Maybe the log extra's (I believe, at least with monolog, a log has a context and an extra object to store structured metadata)?

Reply:
In which case we'd need another way to filter on logs in the web profiler/debug toolbar. Which is not an area I can help 😊